enumerable.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    for element in self
+      yield element
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
```

Comment:
- [ ] When you call each_with_index on ranges if a block is given it should return the range itself. 
Your my_each_with_index returns flatten the range to an array.

`(5..10).my_each_with_index{|num| num*2}` this expression should return `(5..10)` but your implementation returns `[5, 6, 7, 8, 9, 10]`

You can check the type of parameter before converting it to an array.